sys/vm/vm_kern.c

```diff
@@ -657,10 +670,16 @@ kmap_alloc_wait(vm_map_t map, vm_size_t size)
 		map->needs_wakeup = TRUE;
 		vm_map_unlock_and_wait(map, 0);
 	}
-	vm_map_insert(map, NULL, 0, addr, addr + size, VM_PROT_RW_CAP,
-	    VM_PROT_RW_CAP, MAP_ACC_CHARGED, VM_MIN_KERNEL_ADDRESS);
+
+	mapped = addr;
+	if (vm_map_reservation_create_locked(map, &mapped, padded_size,
+	    VM_PROT_RW_CAP))
+		return (0);
```

Comment:
returning with the map locked seems wrong.

Reply:
indeed